File: 3-leetcode/02-medium/js/2924-find-champion-ii.ts
const findChampion = (n: number, edges: number[][]): number => {
  const inDegrees = Array.from({ length: n }, () => 0);

  for (let edge of edges) {
    inDegrees[edge[1]]++;
  }

  let champ = -1;
  let isChampExists = false;

  for (let i = 0; i < n; i++) {
    if (inDegrees[i] != 0) {
      continue;
    }
    if (isChampExists) {
      return -1;
    }
    champ = i;
    isChampExists = true;
  }

  return champ;
};

File: 3-leetcode/02-medium/dart/2924-find-champion-ii.dart
class Solution {
  int findChampion(int n, List<List<int>> edges) {
    List<int> inDegrees = List.generate(n, (i) => 0);

    for (final edge in edges) {
      inDegrees[edge[1]]++;
    }

    int champ = -1;
    bool isChampExists = false;

    for (int i = 0; i < n; i++) {
      if (inDegrees[i] == 0) {
        if (isChampExists) {
          return -1;
        }

        champ = i;
        isChampExists = true;
      }
    }

    return champ;
  }
}
